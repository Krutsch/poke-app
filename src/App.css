* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

.app {
  text-align: center;
  background-color: #282c34;
  font-size: calc(10px + 2vmin);
  color: white;
  min-height: 100vh;
}

.main {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(215px, max-content));
  justify-content: center;
  grid-gap: 2em;
}

.logo {
  width: 100%;
  height: auto;
  max-height: 300px;
  object-fit: contain;
}

.favorites {
  display: block;
  text-decoration: none;
  text-align: left;
  margin-bottom: 1em;
  margin-left: 1em;
  color: whitesmoke;
  transition: transform 0.2s;
  width: 200px; /*EDGE*/
  width: max-content;
}
.favorites:hover {
  transform: scale(1.1);
}
