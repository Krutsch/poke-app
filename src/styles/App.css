.app {
  text-align: center;
  font-size: calc(10px + 2vmin);
  color: white;
  padding-bottom: 1em;
}

.main {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(215px, max-content));
  justify-content: center;
  grid-gap: 2em;
}

.logo {
  width: 100%;
  height: auto;
  max-height: 300px;
  object-fit: contain;
}

.header {
  margin-bottom: 1em;
}

.favorites {
  display: block;
  text-decoration: none;
  text-align: left;
  margin-bottom: 1em;
  margin-left: 1em;
  color: whitesmoke;
  transition: transform 0.2s;
  width: 200px; /*EDGE*/
  width: max-content;
}

.favorites:hover {
  transform: scale(1.05);
}
