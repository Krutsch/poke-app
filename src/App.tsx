import React, { lazy } from "react";
import "./App.css";
import Pokemon from "./Pokemon";
import Searchbar from "./Searchbar";
import Heart from "./Heart";
import { BrowserRouter as Router, Route, Switch, Link } from "react-router-dom";
const Detail = lazy(() => import("./Detail"));
const Favorites = lazy(() => import("./Favorites"));

function App() {
  return (
    <React.StrictMode>
      <div className="app">
        <Router>
          <header className="header">
            <Link to="/">
              <img
                className="logo"
                src="https://upload.wikimedia.org/wikipedia/commons/thumb/9/98/International_Pok%C3%A9mon_logo.svg/2000px-International_Pok%C3%A9mon_logo.svg.png"
                alt="Pokemon"
              />
            </Link>
          </header>

          <Switch>
            <Route exact path="/">
              <Searchbar history={window.history} />
              <Link className="favorites" to="/favorites">
                <Heart class="heart" />
                Favorites
              </Link>
              <main className="main">
                {[...new Array(20).keys()].map(i => (
                  <Pokemon key={++i} identifier={i}></Pokemon>
                ))}
              </main>
            </Route>
            <Route path="/detail" component={WaitingComponent(Detail)} />
            <Route path="/favorites" component={WaitingComponent(Favorites)} />
          </Switch>
        </Router>
      </div>
    </React.StrictMode>
  );
}

function WaitingComponent(Component: React.ElementType) {
  return (props: unknown) => (
    <React.Suspense fallback={<div>Loading...</div>}>
      <Component {...props} />
    </React.Suspense>
  );
}

export default App;
