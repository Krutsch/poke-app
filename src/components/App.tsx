import React, { lazy, Suspense } from "react";
import "../styles/App.css";
import Logo from "../images/logo.webp";
import Pokemon from "./Pokemon";
import Searchbar from "./Searchbar";
import Heart from "./Heart";
//@ts-ignore
// eslint-disable-next-line
import { BrowserRouter as Router, Route, Routes, Link, Outlet } from "react-router-dom";
const lazyDetail = lazy(() => import("../routes/Detail"));
const lazyFav = lazy(() => import("../routes/Favorites"));
const Favorites = WaitingComponent(lazyFav);
const Detail = WaitingComponent(lazyDetail);

function App() {
  return (
    <React.StrictMode>
      <div className="app">
        <Router>
          <header className="header">
            <Link to="/">
              <img className="logo" src={Logo} alt="Pokemon" height="300" width="1700" />
            </Link>
          </header>

          <Routes>
            <Route path="/" element={<EnterPage />} />
            <Route path="detail/*" element={<Detail />} />
            <Route path="favorites" element={<Favorites />} />
          </Routes>
        </Router>
      </div>
    </React.StrictMode>
  );
}

function WaitingComponent(Component: React.ElementType) {
  return (props: unknown) => (
    <React.Suspense fallback={<></>}>
      <Component {...props} />
    </React.Suspense>
  );
}
function EnterPage() {
  return (
    <>
      <Searchbar />
      <Link className="favorites" to="/favorites">
        <Heart class="heart" />
        Favorites
      </Link>
      <main className="main">
        {[...new Array(20).keys()].map(i => (
          <Suspense key={++i} fallback={<></>}>
            <Pokemon identifier={i}></Pokemon>
          </Suspense>
        ))}
      </main>
    </>
  );
}
WaitingComponent.whyDidYouRender = true;
EnterPage.whyDidYouRender = true;
App.whyDidYouRender = true;
export default App;
