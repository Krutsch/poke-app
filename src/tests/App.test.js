import React from "react";
import App from "../components/App";
import ShallowRenderer from "react-test-renderer/shallow";

it("renders correctly", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
