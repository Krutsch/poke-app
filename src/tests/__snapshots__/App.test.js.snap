// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<UNDEFINED>
  <div
    className="app"
  >
    <BrowserRouter>
      <header
        className="header"
      >
        <ForwardRef
          to="/"
        >
          <img
            alt="Pokemon"
            className="logo"
            height="300"
            src="logo.webp"
            width="1700"
          />
        </ForwardRef>
      </header>
      <Switch>
        <Route
          exact={true}
          path="/"
        >
          <Searchbar />
          <ForwardRef
            className="favorites"
            to="/favorites"
          >
            <Heart
              class="heart"
            />
            Favorites
          </ForwardRef>
          <main
            className="main"
          >
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={1}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={2}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={3}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={4}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={5}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={6}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={7}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={8}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={9}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={10}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={11}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={12}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={13}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={14}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={15}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={16}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={17}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={18}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={19}
              />
            </React.Suspense>
            <React.Suspense
              fallback={<React.Fragment />}
            >
              <Pokemon
                identifier={20}
              />
            </React.Suspense>
          </main>
        </Route>
        <Route
          component={[Function]}
          path="/detail"
        />
        <Route
          component={[Function]}
          path="/favorites"
        />
      </Switch>
    </BrowserRouter>
  </div>
</UNDEFINED>
`;
